src/auto-evo/RunResults.cs

```diff
@@ -368,6 +328,155 @@ public long GetPopulationInPatch(Species species, Patch patch)
+        ///   </para>
+        /// </remarks>
+        public Dictionary<Patch, Dictionary<Species, long>> GetPopulationsByPatch(Species species,
+            bool resolveMigrations = false, bool resolveSplits = false, bool includeNewSpecies = true)
+        {
+            var speciesInPatches = new Dictionary<Patch, Dictionary<Species, long>>();
+
+            if (!results.ContainsKey(species))
+            {
+                GD.Print("Species ", species.FormattedName, "not found in results, returning empty dictionary.");
+                return speciesInPatches;
+            }
+
+            var speciesResult = results[species];
+
+            // Get natural variations
+            foreach (var patchPopulationEntry in speciesResult.NewPopulationInPatches)
+            {
+                if (!speciesInPatches.ContainsKey(patchPopulationEntry.Key))
+                    speciesInPatches[patchPopulationEntry.Key] = new Dictionary<Species, long>();
+
+                if (patchPopulationEntry.Value > 0)
+                {
+                    if (includeNewSpecies || speciesResult.NewlyCreated == null)
+                    {
+                        speciesInPatches[patchPopulationEntry.Key].Add(
+                            species, patchPopulationEntry.Value);
+                    }
+                }
+            }
+
+            if (resolveSplits)
+            {
+                // Deal with splits-off
+                if (speciesResult.SplitOff != null && speciesResult.SplitOffPatches != null)
+                {
+                    foreach (var patch in speciesResult.SplitOffPatches)
+                    {
+                        speciesInPatches[patch].Remove(species);
+                        speciesInPatches[patch].Add(speciesResult.SplitOff,
+                            speciesResult.NewPopulationInPatches[patch]);
+                    }
+                }
+            }
+
+            if (resolveMigrations)
+            {
+                // Apply migrations
+                foreach (var migration in speciesResult.SpreadToPatches)
+                {
+                    // Only consider valid migrations
+                    if (migration.From == null || migration.To == null)
+                        continue;
+
+                    // Add entries if necessary
+                    if (!speciesInPatches.ContainsKey(migration.From))
+                        speciesInPatches[migration.From] = new Dictionary<Species, long>();
+
+                    if (!speciesInPatches.ContainsKey(migration.To))
+                        speciesInPatches[migration.To] = new Dictionary<Species, long>();
+
+                    // Only consider possible migrations
+                    // TODO check second condition to mimick actual effect. (> 0 ?)
+                    if (speciesInPatches[migration.From].TryGetValue(species, out var population) &&
+                        population >= migration.Population)
+                    {
```

Comment:
I think this is new? This isn't in my original code because there is a separate step to filter out invalid migrations.
I don't think this checking should be added here...